### ISSUE TITLE
iprintf definition makes compilation errors under Windows (Cygwin)

### ISSUE THREAD

**Issue by Dji75** [esnet/iperf#280]
As already defined under /usr/include/stdio.h, the iiprintf() function defined in iperf_api.h gives compilation errors when compiled under Windows (Cygwin)

**Comment by bmah888** [comment]
Cygwin is not a officially supported platform, and not one I have easy access to.  I had the hazy impression that there were some users running iperf3 under Cygwin already but I could be mistaken.

Any chance you can submit a pull request for this?

**Comment by Dji75** [comment]
I'll try :)

**Comment by tniessen** [comment]
@Dji75 You can safely rename `iprintf` to `iperf_printf`. Some things still require attention, for example, the following code from `iperf_new_stream` ([source](https://github.com/esnet/iperf/blob/a8a902ddf8d4e8dc8abf8829ad253ed4a45ec0d4/src/iperf_api.c#L2609-L2614)) will fail as Windows does not support unlinking files which are held open by a process.

```c
if (unlink(template) < 0) {
    i_errno = IECREATESTREAM;
    free(sp->result);
    free(sp);
    return NULL;
}
```

Also note that some hardcoded paths (`/tmp`) might lead to problems under certain conditions if run outside of the cygwin environment.

**Comment by Dji75** [comment]
I've replace /tmp path to local path
For the unlinking issue, as I'm not c++ expert, I have no idea of the fix.

I've just submitted the pull request : https://github.com/esnet/iperf/pull/295

**Comment by Dji75** [comment]
No news about my pull request ?

**Comment by bmah888** [comment]
Ah sorry.  I've been heads-down on a number of other projects (iperf3 is part of my job, but right now is a small part).  I've got one comment on your pull request...I'll follow-up with a comment there.

**Comment by bmah888** [comment]
Closed via pull request #468.
